python/ray/serve/backend_worker.py

```diff
@@ -48,108 +78,82 @@ def ensure_async(func):
         return sync_to_async(func)
 
 
-class RayServeMixin:
-    """This mixin class adds the functionality to fetch from router queues.
-
-    Warning:
-        It assumes the main execution method is `__call__` of the user defined
-        class. This means that serve will call `your_instance.__call__` when
-        each request comes in. This behavior will be fixed in the future to
-        allow assigning artibrary methods.
-
-    Example:
-        >>> # Use ray.remote decorator and RayServeMixin
-        >>> # to make MyClass servable.
-        >>> @ray.remote
-            class RayServeActor(RayServeMixin, MyClass):
-                pass
-    """
-    _ray_serve_self_handle = None
-    _ray_serve_router_handle = None
-    _ray_serve_setup_completed = False
-    _ray_serve_dequeue_requester_name = None
-
-    # Work token can be unfullfilled from last iteration.
-    # This cache will be used to determine whether or not we should
-    # work on the same task as previous iteration or we are ready to
-    # move on.
-    _ray_serve_cached_work_token = None
-
-    _serve_metric_error_counter = 0
-    _serve_metric_latency_list = []
-
-    def _serve_metric(self):
-        # Make a copy of the latency list and clear current list
-        latency_lst = self._serve_metric_latency_list[:]
-        self._serve_metric_latency_list = []
+class RayServeWorker:
+    """Fetches requests and handles them with the provided callable."""
+
+    def __init__(self,
+                 name,
+                 _callable,
+                 self_handle,
+                 router_handle,
+                 is_function=False):
```

Comment:
this should be positional? it should always be passed in explicitly right? it doesn't make sense to me to assume a value here

Reply:
Fair point, changed